src/screens/ImportWalletScreen.tsx

```diff
@@ -19,88 +19,226 @@ import {
+          try {
+            wallet.addPublicKey(instantPublicKey);
+            this.addRecoveryKey(wallet);
+          } catch (_) {
+            this.showErrorMessageScreen();
+          }
+        },
+        title: i18n.wallets.publicKey.recoverySubtitle,
+        description: i18n.wallets.publicKey.recoveryDescription,
+        withLink: false,
+      });
+    } catch (_) {}
+  };
+
+  saveARWallet = async (wallet: HDSegwitP2SHArWallet, pubKeyHex: string) => {
+    try {
+      wallet.addPublicKey(pubKeyHex);
+      await wallet.generateAddresses();
+      await wallet.fetchBalance();
+      await wallet.fetchTransactions();
+      if (wallet.getBalance() > 0 || wallet.getTransactions().length !== 0) {
+        this.saveWallet(wallet);
+      } else {
+        this.showErrorMessageScreen();
+      }
+    } catch (error) {
+      this.showErrorMessageScreen();
+    }
+  };
+
+  saveAIRWallet = async (wallet: HDSegwitP2SHAirWallet, recoveryPublicKey: string) => {
+    try {
+      wallet.addPublicKey(recoveryPublicKey);
+      await wallet.generateAddresses();
+      await wallet.fetchBalance();
+      await wallet.fetchTransactions();
+      if (wallet.getBalance() > 0 || wallet.getTransactions().length !== 0) {
+        this.saveWallet(wallet);
+      } else {
+        this.showErrorMessageScreen();
+      }
+    } catch (error) {
+      this.showErrorMessageScreen();
+    }
+  };
+
+  renderSubtitle = () => {
+    const { walletType } = this.props?.route.params;
+    if (walletType === HDSegwitP2SHArWallet.type || walletType === HDSegwitP2SHAirWallet.type) {
+      return (
+        <>
+          <View style={styles.arSubtitleContainer}>
+            <Text style={styles.subtitle}>{i18n.wallets.importWallet.importARDescription1}</Text>
+            <Text style={styles.subtitle}>{i18n._.or}</Text>
+            <Text style={styles.subtitle}>{i18n.wallets.importWallet.importARDescription2}</Text>
+          </View>
+          <InputItem
+            error={this.state.validationError}
+            setValue={this.onLabelChange}
+            label={i18n.wallets.add.inputLabel}
+          />
+        </>
+      );
+    }
+    return <Text style={styles.subtitle}>{i18n.wallets.importWallet.subtitle}</Text>;
+  };
+
+  addRecoveryKey = (wallet: HDSegwitP2SHAirWallet) => {
+    this.props.navigation.navigate(Route.IntegrateKey, {
+      onBarCodeScan: (recoveryPublicKey: string) => {
+        CreateMessage({
+          title: i18n.message.creatingWallet,
+          description: i18n.message.creatingWalletDescription,
+          type: MessageType.processingState,
+          asyncTask: () => {
+            this.saveAIRWallet(wallet, recoveryPublicKey);
+          },
+        });
+      },
+      withLink: false,
+      title: i18n.wallets.publicKey.recoverySubtitle,
+      description: i18n.wallets.publicKey.recoveryDescription,
+    });
+  };
+
+  importMnemonic = async (mnemonic: string) => {
+    mnemonic.trim();
```

Comment:
trim() doesn't change the original string, it returns a new one, so this operation does nothing here. You have to assign it to new value like `const mnemonicTrimmed = mnemonic.trim()`